src/psyclone/psyir/tools/call_tree_utils.py

```diff
@@ -0,0 +1,415 @@
+                    # Builtins don't have non-local accesses
+                    continue
+
+                # Get the non-local access information from the kernel
+                # by querying the module that contains the kernel:
+                try:
+                    mod_info = mod_manager.get_module_info(kernel.module_name)
+                except FileNotFoundError:
+                    print(f"Could not find module '{kernel.module_name}' - "
+                          f"ignored.")
+                    continue
+
+                all_possible_routines = mod_info.resolve_routine(kernel.name)
+                for routine_name in all_possible_routines:
+                    psyir = mod_info.get_psyir(routine_name)
+                    todo.extend(self.get_non_local_symbols(psyir))
+        return self._resolve_calls_and_unknowns(todo, read_write_info)
+
+    # -------------------------------------------------------------------------
+    def _resolve_calls_and_unknowns(self, todo, read_write_info):
+        '''This function updates the list of non-local symbols by:
+        1. replacing all subroutine calls with the list of their corresponding
+            non-local symbols.
+        2. Resolving unknown types to be either function calls (which then
+            get resolved as above), or variables.
+        This is done recursively until only variable accesses remain.
+
+        The actual non-local accesses will then be added to the ReadWriteInfo
+        object.
+
+        :param todo: the information about symbol type, module_name,
+            symbol_name and access information
+        :type todo: List[Tuple[str,str, str,\
+                              :py:class:`psyclone.core.Signature`,str]]
+        :param read_write_info: information about all input and output
+            parameters.
+        :type read_write_info: :py:class:`psyclone.psyir.tools.ReadWriteInfo`
+
+        '''
+        # pylint: disable=too-many-branches
+        mod_manager = ModuleManager.get()
+        done = set()
+        # Using a set here means that duplicated entries will automatically
+        # be filtered out.
+        in_vars = set()
+        out_vars = set()
+        while todo:
+            info = todo.pop()
+            if info in done:
+                continue
+            done.add(info)
+            external_type, module_name, signature, access_info = info
+            if module_name in mod_manager.ignores():
+                continue
+            if external_type == "routine":
+                if module_name is None:
+                    # We don't know where the subroutine comes from.
+                    # For now ignore this
+                    print(f"Unknown routine '{signature[0]} - ignored.")
+                    continue
+                try:
+                    mod_info = mod_manager.get_module_info(module_name)
+                except FileNotFoundError:
+                    print(f"Cannot find module '{module_name}' - ignored.")
```

Comment:
#11 here too please. In fact, I think these print statements ought to be removed as they don't have any contextual information and will be confusing to a user. Alternatively, please could you prefix them with e.g. "Attempting to resolve non-local accesses [in xxx] - cannot find ...."

Reply:
Done.